### ISSUE TITLE
New rule: no-greater-than

### ISSUE THREAD

**Issue by janpaul123** [eslint/eslint#10591]
**Please describe what the rule should do:**

Disallow operators `>` and `>=`.

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
if (2 > 1) {
    // ...
}

if (2 >= 1) {
    // ...
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

[See Llewellyn Falco explanation.](http://llewellynfalco.blogspot.com/2016/02/dont-use-greater-than-sign-in.html)

I completely plagiarised this from https://github.com/eslint/eslint/pull/8677 — it would be great to revive that rule, especially since it supports fixing!

**Comment by kaicataldo** [comment]
I agree with the sentiments in https://github.com/eslint/eslint/pull/8677 - this doesn't feel like a great candidate for core, particularly since this is easily achieved with [`no-restricted-syntax`](https://github.com/eslint/eslint/pull/8677#issuecomment-305596751).

**Comment by platinumazure** [comment]
> But no-restricted-syntax can't do autofixing for this.

True enough, but writing a custom rule would allow you to include autofixing.

We currently maintain well over 200 rules, so our policy for new core rules is that we should only accept new rules that are critical. This rule proposal, in my view, does not add critically missing functionality.
